File: ListaEnlazada.cpp
#include <iostream>

using namespace std;

struct Nodo {
    int dato;
    Nodo* siguiente;
};

// Función para insertar elementos al inicio
void insertarInicio(Nodo*& lista, int n) {
    Nodo* nuevo_nodo = new Nodo(); // 1. Crear espacio en memoria
    nuevo_nodo->dato = n;          // 2. Asignar el dato
    nuevo_nodo->siguiente = lista; // 3. Apuntar al antiguo inicio
    lista = nuevo_nodo;            // 4. El nuevo nodo ahora es el inicio
    cout << "Elemento " << n << " insertado correctamente.\n";
}

// Función para mostrar la lista
void mostrarLista(Nodo* lista) {
    Nodo* actual = lista;
    cout << "\nLista completa: ";
    while (actual != nullptr) {
        cout << "[" << actual->dato << "] -> ";
        actual = actual->siguiente;
    }
    cout << "NULL" << endl;
}

int main() {
    Nodo* lista = nullptr; // Inicializamos la lista vacía
    int valor, cantidad;

    cout << "¿Cuantos numeros desea ingresar? ";
    cin >> cantidad;

    for (int i = 0; i < cantidad; i++) {
        cout << "Digite un numero: ";
        cin >> valor;
        insertarInicio(lista, valor);
    }

    mostrarLista(lista);

    return 0;
}

File: ListaEnlazada.py
class Nodo:
    def __init__(self, dato):
        self.dato = dato
        self.siguiente = None  # Equivale al nullptr de C++

class ListaEnlazada:
    def __init__(self):
        self.cabeza = None

    # Insertar al inicio (Equivalente a push en una pila)
    def insertar_inicio(self, dato):
        nuevo_nodo = Nodo(dato)
        nuevo_nodo.siguiente = self.cabeza
        self.cabeza = nuevo_nodo
        print(f"Elemento {dato} insertado.")

    # Recorrer la lista
    def mostrar_lista(self):
        actual = self.cabeza
        print("\nLista actual:", end=" ")
        while actual:
            print(f"[{actual.dato}] ->", end=" ")
            actual = actual.siguiente
        print("None")

def main():
    mi_lista = ListaEnlazada()
    
    # El usuario digita los datos
    try:
        cantidad = int(input("¿Cuántos números desea ingresar a la lista? "))
        
        for i in range(cantidad):
            valor = int(input(f"Digite el valor {i+1}: "))
            mi_lista.insertar_inicio(valor)
        
        mi_lista.mostrar_lista()
    except ValueError:
        print("Error: Por favor ingrese solo números enteros.")

if __name__ == "__main__":
    main()
